File: IS_Lab2/IS_Lab2_C#/Kod źródłowy/Serialize_json.cs

using Newtonsoft.Json;

namespace Lab2
{
    internal class Serialize_json
    {
        public static void run(Deserialize_json deserializeData, string filelocation)
        {
            Console.WriteLine("let's serialize JSON");

            JsonSerializer serializer = new JsonSerializer();
            serializer.NullValueHandling = NullValueHandling.Ignore;

            List<Dictionary<string, dynamic>> data = new List<Dictionary<string, dynamic>>();

            foreach (var dep in deserializeData.data)
            {
                var powiat = "-";
                if (dep.ContainsKey("Powiat"))
                {
                    powiat = dep["Powiat"];
                }

                var record = new Dictionary<string, dynamic>();

                string[] fieldList = {"Kod_TERYT", "Województwo", "Powiat", "typ_JST", "nazwa_urzędu_JST", "miejscowość", "telefon_z_numerem_kierunkowym"};

                foreach(var field in fieldList)
                {
                    if (field == "Powiat")
                        record.Add(field, powiat);
                    else if (field == "telefon_z_numerem_kierunkowym")
                        record.Add(field, (dep["telefon kierunkowy"].ToString() + '-' + dep["telefon"].ToString()));
                    else
                        record.Add(field, dep[field]);
                }
                data.Add(record);
            }

            using (StreamWriter sw = new StreamWriter(filelocation))
            using (JsonWriter writer = new JsonTextWriter(sw))
            {

                serializer.Serialize(writer, data);
            }

            Console.WriteLine("it is done");
        }
    }
}

File: IS_Lab2/IS_Lab2_JSON/Kod źródłowy/serialize_json.py
# -*- coding: utf-8 -*-
"""
seria
"""

import json


class SerializeJson:

    # metoda statyczna
    @staticmethod
    def run(deserializeddata, filelocation):
        print("let's serialize JSON")

        lst = []
        for dep in deserializeddata.data:
            powiat = '-'
            if 'Powiat' in dep:
                powiat = dep['Powiat']
            lst.append({
                "Kod_TERYT": dep['Kod_TERYT'],
                "Województwo": dep['Województwo'],
                "Powiat": powiat,
                "typ_JST": dep['typ_JST'],
                "nazwa_urzędu_JST": dep['nazwa_urzędu_JST'],
                "miejscowość": dep['miejscowość'],
                "telefon_z_numerem_kierunkowym": str(dep['telefon kierunkowy']) + '-' + str(dep['telefon'])
            })
        jsontemp = {"departaments": lst}
        with open(filelocation, 'w', encoding='utf-8') as f:
            json.dump(jsontemp, f, ensure_ascii=False)
        print("it is done\n")
